bemserver_core/model/sites.py

```diff
@@ -1,11 +1,28 @@
 """Sites"""
+import enum
 import sqlalchemy as sqla
 import sqlalchemy.orm as sqlaorm
 from sqlalchemy.ext.hybrid import hybrid_property
 
 from bemserver_core.database import Base
 from bemserver_core.model import Campaign
 from bemserver_core.authorization import AuthMixin, auth, Relation
+from bemserver_core.exceptions import PropertyTypeInvalidError
+
+
+class PropertyType(enum.Enum):
+    integer = int
+    float = float
+    boolean = bool
+    string = str
+
+    def cast(self, val_in):
+        val_out = str(val_in)
+        if self is self.boolean:
+            val_out = val_out.lower() in ["true", "t", "1", "yes", "y"]
```

Comment:
Couldn't we just keep `val_out is True`? Do we actually expect other values to be passed?

Through the JSON API, I don't think so.

Reply:
No, JSON API field is a string.
Just allow "true" (lowered) or "1" for True and "false" (lowered) or "0" for False. Any other value must raise an exception.

Reply:
Allowing a single representation would ensure round-trip (the data that comes out of the database is the same as the data that was entered). I'd go for "true" / "false" only.

The only thing that bothers me is that `str(True)` produces `"True"`, so you won't be able to do `my_obj.my_prop = True`. But this is consistent with the fact that the property should always be considered to be a string and should be cast when used.

You could write

```py
try:
    val_out = {"true": True, "false": False}
except KeyError:
    raise ....
```

In fact, I'm wondering if this method should actually cast or just check castability. No point casting before sending to SQLAlchemy String field.

Reply:
Changed method to check value "castability" instead of cast and return casted value.